library/ssl_msg.c

```diff
@@ -740,12 +783,17 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
         mode == MBEDTLS_MODE_CCM ||
```

Comment:
One more question regarding conversion between mbedtls cipher mode and psa alg.
Translation function determines alg for `CCM` as follows:
`*alg = taglen ? PSA_ALG_AEAD_WITH_SHORTENED_TAG( PSA_ALG_CCM, taglen ) : PSA_ALG_CCM;`

Does `mode == MBEDTLS_MODE_CCM` corresponds to:
```
psa_alg == PSA_ALG_CCM || PSA_ALG_IS_AEAD(psa_alg)
```
if psa crypto is enabled?

Reply:
I think `mode == MBEDTLS_MODE_CCM` is equivalent to `PSA_ALG_IS_AEAD(psa_alg) && PSA_ALG_AEAD_WITH_DEFAULT_LENGTH_TAG( psa_alg ) == PSA_ALG_CCM`. Or in the context of TLS, it's also equivalent to `psa_alg == PSA_ALG_CCM || psa_alg == PSA_ALG_AEAD_WITH_SHORTENED_TAG( PSA_ALG_CCM, 8 )` because TLS only uses the default taglen or 8.

Also, in contexts like this one were the existing code checks or "CCM or GCM or ChachaPoly", then that condition is equivalent to `PSA_ALG_IS_AEAD(psa_alg)` which is much simpler :)